pkg/sanitization/aws/cloud_map.go

```diff
@@ -0,0 +1,17 @@
+package aws
+
+import (
+	"regexp"
+
+	"github.com/klothoplatform/klotho/pkg/sanitization"
+)
+
+// PrivateDnsNamespaceSanitizer returns a sanitized private dns namespace when applied.
+var PrivateDnsNamespaceSanitizer = sanitization.NewSanitizer(
+	[]sanitization.Rule{
+		// strip any leading non alpha characters
+		{
+			Pattern:     regexp.MustCompile(`^[!-~]+$`),
```

Comment:
This rule seems off:
1. `// strip any leading non alpha characters` seems like a copy/paste issue -- that's not what this pattern does
2. `[!-~]` should be `[^!-~]` since this rule is supposed to detect invalid characters for replacement.
3. The max length should be 253 per documentation

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-servicediscovery-privatednsnamespace.html

I don't know how much this matters if we're still planning a resource name validation pass on the codebase at the end.